### PR TITLE
Drop featureNumeric

### PR DESCRIPTION
**Pull request by cocreature**
Also unused now that we only support LF >= 1.14.

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: compiler/damlc/daml-prim-src/GHC/Num.daml [hunk 1]
```diff
@@ -120,22 +120,16 @@ instance Signed Int where
...
-#ifdef DAML_NUMERIC
-#define IF_NUMERIC(a,b) a
 #define NUMERIC_LIT(n) (primitive @"BECastNumeric" ((n) : Numeric 10))
-#else
-#define IF_NUMERIC(a,b) b
-#define NUMERIC_LIT(n) n
-#endif
 
-instance Additive IF_NUMERIC((Numeric n), Decimal) where
-    (+) = primitive @IF_NUMERIC("BEAddNumeric", "BEAddDecimal")
```
Comment (akrmn) [inline compiler/damlc/daml-prim-src/GHC/Num.daml:132]: can we get rid of `BTDecimal` now? Also the primitives `BEAddDecimal` et al.?

**Review by akrmn — APPROVED** [review]
🧹🧹🧹
Thanks!

File: compiler/damlc/daml-prim-src/GHC/Num.daml (hunk 1 above)
Comment (cocreature) [reply compiler/damlc/daml-prim-src/GHC/Num.daml:132]: good point, I think we can but it looks like it’s a larger change so I’ll move it to a separate PR.

Comment (akrmn) [reply compiler/damlc/daml-prim-src/GHC/Num.daml:132]: that sounds good, thanks!

**Pull request merged**